app/services/force/soql/lease_up_application_service.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Force
+  module Soql
+    # Provide Salesforce SOQL API interactions for applications
+    class LeaseUpApplicationService < Force::Base
+      FIELD_NAME = :applications
+      FIELDS = load_fields(FIELD_NAME).freeze
+
+      def lease_up_applications(opts)
+        query_scope = lease_up_applications_query(opts)
+
+        query_scope.query
+      end
+
+      def build_applications_search(search_terms_string)
+        # Given a comma-separated list of search terms:
+        # For each term, we search across name and application number for a match
+        # For multiple terms, we expect the record to match each of the terms individually.
+        search_terms = search_terms_string.split(',')
+        search_fields = [
+          'Name',
+          'Applicant__r.First_Name__c',
+          'Applicant__r.Last_Name__c',
+        ]
+        search = []
+        search_terms.each do |search_term|
+          # For each term we join the fields with OR to return a record if it matches
+          # across any of these fields.
+          search_term_clause = []
+          sanitized_term = ActiveRecord::Base.sanitize_sql_like search_term
+          search_fields.each do |search_field|
+            search_term_clause.push("#{search_field} like '%#{sanitized_term}%'")
+          end
+          search.push("(#{search_term_clause.join(' OR ')})")
+        end
+        # If there are multiple terms, join the clauses with "AND" because we expect each term to find a match.
+        search.join(' AND ')
+      end
+
+      def build_applications_filters(opts)
+        # This builds the syntax need for soql to do an
+        # OR statement when passing the option of `Vision/Hearing`
+        # https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_querying_multiselect_picklists.htm
+        filters = ''
+        if opts[:accessibility].present?
+          accessibility_string = opts[:accessibility].map do |accessibility|
+            accessibility.split(', ').map { |item| "'#{item}'" }
+          end.join(', ')
+          filters += " and Has_ADA_Priorities_Selected__c INCLUDES (#{accessibility_string}) "
```

Comment:
Good q! In preference_service, we are querying preferences, so when a user searches, their searches and filters are supposed to apply to the application, but in the backend, we still do the query on preferences, so we need index into the associated application and then filter/search on those

In lease_up_application_service, we are now querying applications directly (because the fcfs listings don't have preferences), so we don't have to index into the application object from the preference anymore - we are already starting with application

Old query:
Grab all preference objects, index into associated application, apply searches and filters

New query:
Grab all applications, apply searches and filters

Reply:
Thank you! I must have glossed over the main table used in `builder.from(...)`, which would have clued me in. I appreciate your work with our homegrown soql orm 😅